lib/init.coffee

```diff
@@ -41,10 +42,20 @@ module.exports =
             while (match = regex.exec(result.stdout)) isnt null
               line = parseInt(match[1]) or 0
               toReturn.push({
-                type: 'Warning'
+                type: self.getMessageType(match[2])
                 text: match[2]
                 filePath
                 # make range the full line
                 range: helpers.rangeFromLineNumber(textEditor, line - 1)
               })
           return toReturn
+
+  getMessageType: (line) ->
+    if (
+      line.indexOf('used') isnt -1 or
+      line.indexOf('redefines') isnt -1 or
+      line.indexOf('shadowed') isnt -1 or
+      line.indexOf('may be') isnt -1
+    )
+      return 'Warning'
+    return 'Error'
```

Comment:
(It would be too nice if GitHub would actually should this when viewing a file and not only in the diff view...)